### PR TITLE
fix: sync google group attachment files to links

### LINKED ISSUE #232: Sync attachment files from Google Group to forum
Continue: https://github.com/casbin/casnode/issues/112

We can sync a Google Group with a Casnode node now. But date, attachment files are not working yet. We need a fix.

### PR DESCRIPTION
**Pull request by WindSpiritSR**
Fix: https://github.com/casbin/casnode/issues/232

Sync file info to topic end

Maybe we can add a config to configure:
1. Sync files directly to the server `more server space used`
2. Use the server as a proxy server `if google group is private`
3. Only sync attachment files info to a link `maybe we can get more file info like preview and size to make a card`

Signed-off-by: WindSpiritSR <simon343riley@gmail.com>

### CODE REVIEW

**Comment by hsluoyz** [comment]
@Kininaru

**Comment by WindSpiritSR** [comment]
Deployed at: https://forum.leviatan.cn/go/1002
Google Group: https://groups.google.com/g/forum-dev

File: object/mailing_list.go [hunk 1]
```diff
@@ -47,6 +47,15 @@ func (n Node) SyncFromGoogleGroup() {
 		if AuthorMember == nil {
 			continue
 		}
+
+		for msgIndex, msgItem := range messages {
+			if msgItem.Files != nil {
+				for _, msgFileItem := range msgItem.Files {
+					messages[msgIndex].Content = fmt.Sprintf("%s<br>[%s](%s)", messages[msgIndex].Content, msgFileItem.FileName, msgFileItem.Url)
```
Comment (keainye) [inline object/mailing_list.go:54]: I think using `msgItem` is more elegant than `messages[msgIndex]`.

**Review by keainye — CHANGES_REQUESTED** [review]
`go.mod` and `go.sum` is not updated. Please update `google-groups-crawler` to `v0.1.3`.

File: object/mailing_list.go (hunk 1 above)
Comment (WindSpiritSR) [reply object/mailing_list.go:54]: At line 54, each file link will be appended at the end of the `messages[msgindex].Content`

Comment (WindSpiritSR) [reply object/mailing_list.go:54]: or like this?
```go
for msgIndex, msgItem := range messages {
	if msgItem.Files != nil {
		for _, msgFileItem := range msgItem.Files {
			msgItem.Content = fmt.Sprintf("%s<br>[%s](%s)", msgItem.Content, msgFileItem.FileName, msgFileItem.Url)
		}
		messages[msgIndex].Content = msgItem.Content
	}
}
```

Comment (keainye) [reply object/mailing_list.go:54]: @WindSpiritSR Oh, sorry, I forgot that this is not a pointer array, then these codes are no problem, no need to change.  
Please squash your 2 commits then this PR will be ok.

Comment (WindSpiritSR) [reply object/mailing_list.go:54]: @Kininaru It’s okay, I made the same mistake when I wrote it ahhhh
Already squashed

**Review by keainye — APPROVED** [review]
LGTM

**Pull request merged**